src/main/basecamp.ts

```diff
@@ -0,0 +1,150 @@
+import { BASECAMP_CLIENT_ID, BASECAMP_CLIENT_SECRET } from "../../config/environmentVariables";
+
+const BASECAMP_AUTH_URL = 'https://launchpad.37signals.com/authorization/new?type=web_server';
+const BASECAMP_TOKEN_URL = 'https://launchpad.37signals.com/authorization/token?type=web_server';
+const BASECAMP_AUTH_CHECK_URL = 'https://launchpad.37signals.com/authorization.json';
+const OAUTH_BASECAMP_SERVICE_NAME = 'Basecamp';
+const OAUTH_CALLBACK_FUNCTION_NAME = 'oauthCallback'
+const BASECAMP_UNAUTH_ERROR_MSG = 'Basecamp not authenticated. Please try again.';
+const AUTH_DIALOG_MSG = 'You are disconnected from Basecamp. Copy the following link on a new tab to be authorized:\n\n';
+const AUTH_SUCCESS_HTML = 'Connection with Basecamp was successful! You can can close this tab and re-run the program.';
+const AUTH_FAIL_HTML = 'Connection denied. Please close this tab and try again.';
+const LINK_HEADER_MISSING_ERROR_MSG = 'Next URL from Link header is undefined';
+
+const HEADER_AUTHORIZATION = 'Authorization';
+const HEADER_BEARER_TOKEN = 'Bearer ';
+const HEADER_USER_AGENT = 'User-Agent';
+const HEADER_USER_AGENT_NAME = 'Google App Script Onestop Basecamp Integration';
+const HEADER_CONTENT_TYPE = 'Content-Type';
+const HEADER_JSON_CONTENT_TYPE = 'application/json';
+
+const HTTP_POST_METHOD = 'post';
+const HTTP_PUT_METHOD = 'put'
+const HTTP_GET_METHOD = 'get'
+
+export function sendBasecampPostRequest(requestUrl: string, requestPayload: Record<string, any>): Record<string, any> {
+    const response: GoogleAppsScript.URL_Fetch.HTTPResponse = UrlFetchApp.fetch(requestUrl, {
```

Comment:
Not required but we can use a type alias like `type HTTPResponse = GoogleAppsScript.URL_Fetch.HTTPResponse` at the beginning of this file so the `HTTPResponse` type can be used for better readability

Reply:
nice didn't know i could do that, made the change